src/app/core/components/add-question-form/add-question-form.component.ts

```diff
@@ -0,0 +1,115 @@
+import {Component, OnInit} from '@angular/core';
+import {FormArray, FormBuilder, FormControl, FormGroup, FormGroupDirective, NgForm, Validators} from '@angular/forms';
+import {ErrorStateMatcher} from '@angular/material';
+import {AddAnswerClickCountEnum, AnswersArrLengthEnum} from '../../enums';
+
+export class MyErrorStateMatcher implements ErrorStateMatcher {
+  isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+    return !!(control && control.invalid && (control.dirty || control.touched));
+  }
+}
+
+@Component({
+  selector: 'app-add-question-form',
+  templateUrl: './add-question-form.component.html',
+  styleUrls: ['./add-question-form.component.scss']
+})
+export class AddQuestionFormComponent implements OnInit {
+  questionForm: FormGroup;
+  matcher = new MyErrorStateMatcher();
+  clickCount = 0;
+  text = '';
+  arrAnswerLength = AnswersArrLengthEnum.LENGTH;
+  answerClickCount = AddAnswerClickCountEnum.CLICK_COUNT;
+
+  subjects = [
+    {value: 'history', viewValue: 'History'},
+    {value: 'math', viewValue: 'Math'},
+    {value: 'language', viewValue: 'Language'}
+  ];
+
+  levels = [
+    {value: 'low', viewValue: 'Low'},
+    {value: 'medium', viewValue: 'Medium'},
+    {value: 'height', viewValue: 'Height'}
+  ];
+
+  themes = [
+    {value: '1', viewValue: '1'},
+    {value: '2', viewValue: '2'},
+    {value: '3', viewValue: '3'}
+  ];
+
+  // answers = [
+  //   {valid: 'false', label: ''},
+  //   {valid: 'false', label: ''},
+  //   {valid: 'false', label: ''},
+  //   {valid: 'false', label: ''},
+  // ];
+
+  constructor(private fb: FormBuilder) {
+  }
+
+  ngOnInit() {
+    this.questionForm = this.fb.group({
+      subject: this.fb.control(null, Validators.required),
+      theme: this.fb.control(null, Validators.required),
+      level: this.fb.control(null, Validators.required),
+      tags: this.fb.array([], Validators.required),
+      question: this.fb.control(null, Validators.required),
+      answersForm: this.fb.array([])
+    });
+    // this.patch();
+  }
+
+  // patchValues(label, valid) {
+  //   return this.fb.group({
+  //     label: [label],
+  //     valid: [false]
+  //   });
+  // }
+  //
+  //
+  // patch() {
+  //   const control = this.questionForm.get('answersForm') as FormArray;
+  //   this.answers.forEach(x => {
+  //     control.push(this.patchValues(x.label, x.valid));
+  //   });
+  // }
+
+  newTag(tag) {
+    this.text = tag.target.value;
+  }
+
+  removeAnswer(answer) {
+    const control = this.questionForm.get('answersForm') as FormArray;
+    const index = control.value.indexOf(answerToRemove => answerToRemove === answer);
+    control.removeAt(index);
+    this.clickCount--;
+  }
+
+  addAnswer() {
+    if (this.clickCount < this.answerClickCount) {
```

Comment:
if (this.clickCount >= this.answerClickCount) { return }
// other actions

Not need if else. You can use just if